internal/pkg/parser/testcase.go

```diff
@@ -110,23 +114,39 @@ func parseTestCases(files []string) ([]*TestCase, error) {
 			return []*TestCase{}, fmt.Errorf("reading file '%s' failed: %w", file, err)
 		}
 
-		// we need to transform yaml to json so the marshaler from istio works
-		jsonBytes, err := yaml.YAMLToJSON(fileContent)
-		if err != nil {
-			return []*TestCase{}, fmt.Errorf("yamltojson conversion failed for file '%s': %w", file, err)
-		}
+		decoder := yamlV3.NewDecoder(strings.NewReader(string(fileContent)))
 
-		yamlFile := &TestCaseYAML{}
-		err = json.Unmarshal(jsonBytes, yamlFile)
-		if err != nil {
-			return []*TestCase{}, fmt.Errorf("unmarshaling failed for file '%s': %w", file, err)
-		}
+		for {
+			var testcaseInterface interface{}
 
-		if len(yamlFile.TestCases) == 0 {
-			continue
-		}
+			if err = decoder.Decode(&testcaseInterface); err != nil {
+				if errors.Is(err, io.EOF) {
+					break
+				}
+
+				log.Debugf("error while trying to unmarshal into interface", zapcore.Field{Key: "file", Type: zapcore.StringType, String: file})
+				return out, fmt.Errorf("error while trying to unmarshal into interface (%s): %w", file, err)
+			}
 
-		out = append(out, yamlFile.TestCases...)
+			jsonBytes, err := json.Marshal(testcaseInterface)
+			if err != nil {
+				return []*TestCase{}, fmt.Errorf("yamltojson conversion failed for file '%s': %w", file, err)
+			}
+
+			yamlFile := &TestCaseYAML{}
+			err = json.Unmarshal(jsonBytes, yamlFile)
+			if err != nil {
+				log.Debugf("unmarshaling failed for file '%s': %w", file, err)
+				continue
```

Comment:
To me it feels like we should return with an error here instead of continuing. (fail early, fail loud)